### PR TITLE
Develop v1.0

### PR DESCRIPTION
**Pull request by OrientCue**
## v.1.0. Basic RSS Reader
- Develop basic RSS reader that will display the list of topics from "http://news.tut.by/rss/index.rss" into a table view.
- Selecting the item will launch it in the web browser using ` [[UIApplication sharedApplication] openURL:…]`

### CODE REVIEW

File: RSSReader/RSSReader/FeedTableViewController/FeedTableViewController.m [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+  [super dealloc];
+}
+
+#pragma mark - UIViewController Lifecycle
+
+- (void)viewDidLoad {
+  [super viewDidLoad];
+  UINib *nib = [UINib nibWithNibName:NSStringFromClass([AtomItemTableViewCell class]) bundle:nil];
+  [self.tableView registerNib:nib forCellReuseIdentifier:AtomItemTableViewCell.identifier];
+  self.title = @"Tut.by";
```
Comment (trotnic) [inline RSSReader/RSSReader/FeedTableViewController/FeedTableViewController.m:39]: - hardcoded title in navigation bar

File: RSSReader/RSSReader/Presenter/FeedPresenter/FeedPresenter.m [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+#pragma mark - FeedPresenterType
+
+- (void)fetch {
+  [self.view showLoading];
+  __block typeof(self) weakSelf = self;
+  [self.service fetchFeed:^(NSArray<AtomFeedItem *> *items) {
+    [weakSelf.view hideLoading];
+    dispatch_async(dispatch_get_main_queue(), ^{
+      [weakSelf.view appendItems:items];
+    });
```
Comment (trotnic) [inline RSSReader/RSSReader/Presenter/FeedPresenter/FeedPresenter.m:41]: - switching on the main thread is not in the view controller
- ui-related methods are simultaneously called on different threads

Comment (OrientCue) [reply RSSReader/RSSReader/Presenter/FeedPresenter/FeedPresenter.m:41]: Thanks!

File: RSSReader/RSSReader/FeedTableViewController/FeedTableViewController.m (hunk 1 above)
Comment (OrientCue) [reply RSSReader/RSSReader/FeedTableViewController/FeedTableViewController.m]: thanks!

File: RSSReader/RSSReader/FeedTableViewController/FeedTableViewController.m [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+  [super viewDidLoad];
+  self.title = kFeedTitle;
+  [self configureTableView];
+  [self.presenter fetch];
+}
+
+#pragma mark - Configure Table View
+
+- (void)configureTableView {
+  [self.tableView registerNibForCellClasses:@[[AtomItemTableViewCell class]]];
```
Comment (AndreiChernyshou) [inline RSSReader/RSSReader/FeedTableViewController/FeedTableViewController.m:49]: Please, move the tableview creation and configuration into the getter

File: RSSReader/RSSReader/Model/AtomFeedItem.m [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+NSString *const kPubDateKey = @"pubDate";
+
+@implementation AtomFeedItem
+
+- (instancetype)initWithTitle:(NSString *)title
+           articleDescription:(NSString *)articleDescription
+                         link:(NSURL *)link
+                      pubDate:(NSDate *)pubDate {
+  if (self = [super init]) {
+    _title = title ? [title copy] : [@"" copy];
```
Comment (AndreiChernyshou) [inline RSSReader/RSSReader/Model/AtomFeedItem.m:25]: What for do use [@"" copy]?

File: RSSReader/RSSReader/Model/PubDateHelpers/NSDateFormatter+AtomItemPubDate.m [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+#import "NSDateFormatter+AtomItemPubDate.h"
+
+NSString *const kDateFormatAtomOutput = @"dd.MM.yyyy HH:mm";
+NSString *const kDateFormatAtomInput = @"EE, d MMM yyyy HH:mm:ss Z";
+
+@implementation NSDateFormatter (AtomItemPubDate)
+
++ (instancetype)formatterWithDateFormat:(NSString *)dateFormat {
+  NSDateFormatter *df = [NSDateFormatter new];
```
Comment (AndreiChernyshou) [inline RSSReader/RSSReader/Model/PubDateHelpers/NSDateFormatter+AtomItemPubDate.m:16]: Please, use full naming without cut. 'df' is a bad option.

**Review by AndreiChernyshou — COMMENTED** [review]
Please, fix comments

File: RSSReader/RSSReader/FeedTableViewController/FeedTableViewController.m (hunk 3 above)
Comment (OrientCue) [reply RSSReader/RSSReader/FeedTableViewController/FeedTableViewController.m:49]: I did not create there UITableView. This view controller enherited from UITableViewController and you are suggest to override tableView property?

File: RSSReader/RSSReader/Model/AtomFeedItem.m (hunk 4 above)
Comment (OrientCue) [reply RSSReader/RSSReader/Model/AtomFeedItem.m:25]: I thing about initialize title property with empty string instead nil (non optional value, swift backgoround;)). In objc its ok to set nil, thanks!

File: RSSReader/RSSReader/Model/PubDateHelpers/NSDateFormatter+AtomItemPubDate.m (hunk 5 above)
Comment (OrientCue) [reply RSSReader/RSSReader/Model/PubDateHelpers/NSDateFormatter+AtomItemPubDate.m:16]: Ok, thanks!

**Pull request merged**